### PR TITLE
chore: updated query to use index slot index instead

### PR DESCRIPTION
**Pull request by nemo83**

### CODE REVIEW

File: src/main/java/org/cardanofoundation/explorer/api/repository/ledgersyncagg/AddressTxAmountRepository.java [hunk 1]
```diff
@@ -198,8 +198,11 @@ List<AddressTxAmount> findAllByStakeAddressAndTxHashIn(
   @Query(
       value =
           """
-              SELECT max(atm.blockTime) FROM AddressTxAmount atm
-              WHERE atm.unit = :unit
+              WITH latest_token_slot AS ( SELECT MAX(slot) AS slot FROM AddressTxAmount WHERE unit = :unit )
+              SELECT max(blockTime)
+              FROM AddressTxAmount
+              WHERE unit = :unit
+              AND slot = (SELECT slot FROM latest_token_slot)
```
Comment (Sotatek-DucPhung) [inline src/main/java/org/cardanofoundation/explorer/api/repository/ledgersyncagg/AddressTxAmountRepository.java:205]: How about we select the maximum token slot and then use the java CardanoConverter to convert the slot to block_time ?

Comment (nemo83) [reply src/main/java/org/cardanofoundation/explorer/api/repository/ledgersyncagg/AddressTxAmountRepository.java:205]: yup, makes perfect sense

Comment (nemo83) [reply src/main/java/org/cardanofoundation/explorer/api/repository/ledgersyncagg/AddressTxAmountRepository.java:205]: This should work, right?
`AddressTxAmount findTopByUnitOrderBySlotDesc(String unit);`

Comment (Sotatek-DucPhung) [reply src/main/java/org/cardanofoundation/explorer/api/repository/ledgersyncagg/AddressTxAmountRepository.java:205]: Let me check in my side and update code back to you :D

Comment (Sotatek-DucPhung) [reply src/main/java/org/cardanofoundation/explorer/api/repository/ledgersyncagg/AddressTxAmountRepository.java]: Oops, you already did :D

Comment (Sotatek-DucPhung) [reply src/main/java/org/cardanofoundation/explorer/api/repository/ledgersyncagg/AddressTxAmountRepository.java]: new code make sence to me

**Pull request merged**